### DISCUSSION TITLE
Not Getting Data

### DISCUSSION THREAD

**Discussion by srench01** [esphome-econet/esphome-econet#271] [Q&A] [answered]
It appears that my Atom Lite (M5Stack C008) ESP32 is online and functional.  Great.  It is connected to the ATOM RS485 (A131) - and I have now tried two different phone cables - cut and have Black to G, Red to A and Green to B.

No readings are coming into Home Assistant.  

ESPHome sees the device - it is online.

What I do see is after I connect to the device - I see error messages.  I am thinking it is trying to get data over the serial connection and it's not working (see log file).

I've included my econet-hpwh.yaml file (with extension changed to .txt so I could upload).

Hotwater heater is a Rheem Prestige Professions 80 Gallon (PROPH80) similar to the one used in the video.

Any thoughts or ideas?  Any way to test the RS485 to see that it in fact is functioning properly?  

Not a developer - but I sure can copy and paste.

Thanks in advance for any help.

[logs_econet-hpwh_logs.txt](https://github.com/esphome-econet/esphome-econet/files/13710893/logs_econet-hpwh_logs.txt)
[econet-hpwh.txt](https://github.com/esphome-econet/esphome-econet/files/13710944/econet-hpwh.txt)

**Comment by barndawgie** [comment]
Did you install from the Web Tools or some other mechanism? Have you confirmed your wiring matches the diagrams on our wiki (not by color but by pin)?

**Reply by barndawgie** [reply]
Also, I'm curious - do you see that error if not connected to the water heater?

**Comment by srench01** [comment]
Initial install performed from the web install from this url: https://esphome-econet.github.io/install/

I did that multiple times - and then after I watched FIRST youtube video I saw that he modified the YAML to include additional information that mine did not have.  

I'm happy to blow away the ESPHome Folder, The ESPHome device, and even ESPHome app and start all over if that might help.

Regarding wiring - I've attached an image (it is a normal phone cable and does have all 4 copper pins - photo is blurry and appears as 3 - but there are 4 pins.  In this photo the tab is underneath.  

I am also attaching a photo that better represents what I have - although I only have the 4-pin connector and not the 6-pin.

In looking at this more closely - I think my phone cable is in reverse order compared to the image in the project page.

I think perhaps I need Yellow to go to G - Green to A and Red to B.  Would you agree?

![IMG_0361](https://github.com/esphome-econet/esphome-econet/assets/147949143/1e37b8fc-42af-4565-b965-d6b3c3976b40)

![Screenshot 2023-12-19 at 9 40 09 AM](https://github.com/esphome-econet/esphome-econet/assets/147949143/5375b8a9-b166-4e29-a994-ccbec898e34a)

**Reply by barndawgie** [reply]
The key thing to look at is the pins rather than the colors.  Assuming your pic is looking down at the side of the plug with the release tab on it yours looks [the same as mine](https://raw.githubusercontent.com/esphome-econet/econet-docs/main/photos-and-screenshots/Correctly-Wired-K045.jpeg) which means the wiring looks correct.

That said, wiring is the most common cause of getting no data.

Try setting your logging to DEBUG by adding this substitution in your YAML:
` logger_level: DEBUG`

Then can you post logs both with the device disconnected from the heater and then with it connected?

**Reply by srench01** [reply, marked as answer]
It was the wiring.  My phone cables are exactly opposite of yours.  

Correct wiring for me was in fact... Yellow to G - Green to A and Red to B

THANK YOU for the help and pointing me in the right direction

**Comment by barndawgie** [comment]
I will add a note to the wiki as this is the most common issues we see with new installs.
